### PR TITLE
Linked list tests

### PR DESCRIPTION
**Pull request by Jeroenvh99**
I wrote some basic tests for the linked list functions, tbh it was more to practice a bit with testing

### CODE REVIEW

File: tests/tests.cpp [hunk 1]
```diff
@@ -15,12 +15,6 @@
 // For the implementation of the fixture see tests/windowFixture.hpp
 // --------------------------------------------
 
-TEST_F(Window, Basic)
-{
-	// Basic window is already tested in the fixture
-}
-
-
```
Comment (W2Wizard) [inline tests/tests.cpp:23]: This is testing the constructor

File: tests/listtest.c
Comment (W2Wizard) [inline tests/listtest.c:1]: Overall, I sort of get it but also this is not how the testing framework is supposed to be used.

Please take a look at the `test.cpp` file so see how asserts are being used.

**Review by W2Wizard — CHANGES_REQUESTED** [review]
I'd honestly much rather have a resizable array over linked lists one day so I am questioning the need of the linked lists themselves.

Also the need to the test the lists is not really that important more so in relation to images. Say, we add a image we delete a image and expect the list to look a certain way.

File: tests/tests.cpp (hunk 1 above)
Comment (Jeroenvh99) [reply tests/tests.cpp:23]: In that case it shouldn't be removed, excuse me

**Comment by Jeroenvh99** [comment]
Well, this was more of an exercise for myself to practice different testing techniques, so it doesn't really add a lot of value to MLX I guess

**Comment by W2Wizard** [comment]
For now I won't merge this as mentioned for the reasons above.

**Comment by Jeroenvh99** [comment]
That's fair, it doesn't really add much value to MLX as it is

**Pull request closed without merging**